RFS/src/test/java/com/rfs/common/RestClientTest.java

```diff
@@ -0,0 +1,96 @@
+package com.rfs.common;
+
+import com.rfs.tracing.RfsContexts;
+import com.rfs.tracing.TestContext;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.opensearch.migrations.testutils.HttpRequestFirstLine;
+import org.opensearch.migrations.testutils.SimpleHttpResponse;
+import org.opensearch.migrations.testutils.SimpleNettyHttpServer;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Slf4j
+class RestClientTest {
+    @Test
+    public void testGetEmitsInstrumentation() throws Exception{
+        var rootContext = TestContext.withAllTracking();
+        try (var testServer = SimpleNettyHttpServer.makeServer(false, null,
+                this::makeResponseContext)) {
+            var restClient = new RestClient(new ConnectionDetails("http://localhost:" + testServer.port, null, null));
+            try (var topScope = rootContext.createSnapshotCreateContext()) {
```

Comment:
Creating the scope in this case doesn't seem as useful since you only operating inside a single scope.  If we wanted to 'prove' that the scopes were propigated, lets add a second scope where postAsync is used that could be differentiate on, what do you think?